pkg/storage/ingest/reader.go

```diff
@@ -314,16 +314,18 @@ type partitionCommitter struct {
 	logger log.Logger
 }
 
+const noOffset = -100 // using 0, -1, or -2 would be ambiguous with the semantics of the Kafka offset
```

Comment:
Is this really required? An offset can't be negative. We could just check if the offset to commit is negative, and initialise it with -1.